GitCommands/UserRepositoryHistory/LocalRepositoryManager.cs

```diff
@@ -0,0 +1,129 @@
+        /// <exception cref="NotSupportedException"><paramref name="repositoryPath"/> is a URL.</exception>
+        [ContractAnnotation("repositoryPath:null=>halt")]
+        public Task<IList<Repository>> AddAsMostRecentAsync(string repositoryPath)
+        {
+            if (string.IsNullOrWhiteSpace(repositoryPath))
+            {
+                throw new ArgumentException(nameof(repositoryPath));
+            }
+
+            if (PathUtil.IsUrl(repositoryPath))
+            {
+                // TODO: throw a specific exception
+                throw new NotSupportedException();
+            }
+
+            repositoryPath = repositoryPath.Trim()
+                                           .ToNativePath()
+                                           .EnsureTrailingPathSeparator();
+            return AddAsMostRecentRepositoryAsync(repositoryPath);
+
+            Task<IList<Repository>> AddAsMostRecentRepositoryAsync(string path)
+            {
+                return Task.Run<IList<Repository>>(async () =>
+               {
+                   var repositoryHistory = (await LoadHistoryAsync()).ToList();
+
+                   var repository = repositoryHistory.FirstOrDefault(r => r.Path.Equals(path, StringComparison.CurrentCultureIgnoreCase));
+                   if (repository != null)
+                   {
+                       repositoryHistory.Remove(repository);
+                   }
+                   else
+                   {
+                       repository = new Repository(path);
+                   }
+
+                   repositoryHistory.Insert(0, repository);
+
+                   await SaveHistoryAsync(repositoryHistory);
+
+                   return repositoryHistory;
+               });
+            }
+        }
+
+        /// <summary>
+        /// Loads the history of local git repositories from a persistent storage.
+        /// </summary>
+        /// <returns>The history of local git repositories.</returns>
+        public Task<IList<Repository>> LoadHistoryAsync()
+        {
+            int size = AppSettings.RecentRepositoriesHistorySize;
+            return Task.Run<IList<Repository>>(() =>
+            {
+                var history = _repositoryStorage.Load(KeyRecentHistory);
+                if (history == null)
+                {
+                    return Array.Empty<Repository>();
+                }
+
+                return AdjustHistorySize(history, size).ToList();
+            });
+        }
+
+        /// <summary>
+        /// Removes <paramref name="repository"/> from the history of local git repositories in a persistent storage.
+        /// </summary>
+        /// <param name="repository">A repository to remove.</param>
+        /// <returns>An awaitable task.</returns>
+        public Task RemoveFromHistoryAsync(Repository repository)
+        {
+            // TODO:
+            return Task.CompletedTask;
```

Comment:
:bulb: You can do a no-op await like this so the method isn't empty:

```csharp
await Task.CompletedTask;
```

Reply:
Thanks. This method is something I am yet to implement...